lib/core/sdam/server.js

```diff
@@ -240,6 +240,12 @@ class Server extends EventEmitter {
     }
 
     wireProtocol.command(this, ns, cmd, options, (err, result) => {
+      if (err && err instanceof MongoNetworkError) {
```

Comment:
`if (err && err instanceof MongoNetworkError && options.session)`

Reply:
or flatten it out with the lines below:

```js
if (err) {
  if (err instanceof MongoNetworkError && options.session) {
  }
  if (isSDAMUnrecoverableError(err) {
  }
}
```

Reply:
done